airbyte-workers/src/main/java/io/airbyte/workers/ApplicationInitializer.java

```diff
@@ -317,7 +317,7 @@ private void registerSync(final WorkerFactory factory, final MaxWorkersConfig ma
       final Worker syncWorker = factory.newWorker(taskQueue,
           getWorkerOptions(maxWorkersConfig.getMaxSyncWorkers()));
       final WorkflowImplementationOptions options = WorkflowImplementationOptions.newBuilder()
-          .setFailWorkflowExceptionTypes(NonDeterministicException.class).build();
+          .setFailWorkflowExceptionTypes(NonDeterministicException.class, RuntimeException.class).build();
```

Comment:
Any reason to not just fail any exceptions?

Reply:
Found a thread about this, seems like Temporal generally recommends against this because typically, workflow failures are something to be avoided: https://community.temporal.io/t/throwing-exception-vs-failure-in-workflow/1059

But Airbyte tends to prefer explicit workflow failures instead of retries, so I wonder if we have a good reason to go against their recommendation?